message/consumer.go

```diff
@@ -31,41 +25,42 @@ type Consumer struct {
 	closed          chan bool
 	ctx             context.Context
 	cancel          context.CancelFunc
-	consumer        KafkaConsumer
-	messageReciever Reciever
+	consumer        kafka.IConsumerGroup
+	messageReceiver Receiver
 	defaultShutdown time.Duration
 }
 
 // NewConsumer create a NewInstance event consumer.
-func NewConsumer(consumer KafkaConsumer, messageReciever Reciever, defaultShutdown time.Duration) Consumer {
+func NewConsumer(consumer kafka.IConsumerGroup, messageReceiver Receiver, defaultShutdown time.Duration) Consumer {
 	ctx, cancel := context.WithCancel(context.Background())
 
 	return Consumer{
 		closed:          make(chan bool, 1),
 		ctx:             ctx,
 		cancel:          cancel,
 		consumer:        consumer,
-		messageReciever: messageReciever,
+		messageReceiver: messageReceiver,
 		defaultShutdown: defaultShutdown,
 	}
 }
 
-// Listen poll the KafkaConsumer for incoming messages and pass onto the Reciever
+// Listen poll the KafkaConsumer for incoming messages and pass onto the Receiver
 func (c Consumer) Listen() {
 	go func() {
 		defer func() {
 			// Notify that the consumer loop has closed.
 			c.closed <- true
 		}()
 
+		logData := log.Data{"package": packageName}
 		for {
 			select {
-			case consumedMessage := <-c.consumer.Incoming():
-				loggerC.Info("consumer.incoming receieved a message", nil)
-				c.messageReciever.OnMessage(consumedMessage)
+			case consumedMessage := <-c.consumer.Channels().Upstream:
+				log.Event(c.ctx, "consumer.incoming received a message", log.INFO, logData)
+				c.messageReceiver.OnMessage(consumedMessage)
 				c.consumer.Release()
```

Comment:
Should this be changed to commitAndRelease ?

Reply:
At the moment we commit the message as part of the `OnMessage()` only if there is no error and then we do the release here.

kafka_message_receiver.go:
```
// OnMessage unmarshal the kafka message and pass it to the InstanceEventHandler any errors are sent to the ErrorReporter
func (r KafkaMessageReceiver) OnMessage(message kafka.Message) {
	var newInstanceEvent event.NewInstance
	// This context will come from the received kafka message
	ctx := context.Background()
	logData := log.Data{"package": "message.KafkaMessageReceiver"}
	if err := schema.NewInstanceSchema.Unmarshal(message.GetData(), &newInstanceEvent); err != nil {
		log.Event(ctx, "error while attempting to unmarshal kafka message into event.NewInstance", log.ERROR, log.Error(err), logData)
		return
	}

	logData["event"] = newInstanceEvent
	log.Event(ctx, "successfully unmarshalled kafka message into event.NewInstance", log.INFO, logData)

	if err := r.InstanceHandler.Handle(ctx, newInstanceEvent); err != nil {
		log.Event(ctx, "InstanceHandler.Handle returned an error", log.ERROR, log.Error(err), logData)
		if err := r.ErrorReporter.Notify(newInstanceEvent.InstanceID, "InstanceHandler.Handle returned an unexpected error", err); err != nil {
			log.Event(ctx, "ErrorReporter.Notify returned an error", log.ERROR, log.Error(err), logData)
		}
		return
	}

	log.Event(ctx, "newInstance event successfully processed", log.INFO, logData)
	message.Commit()
}
```